src/assets/AssetsContractController.ts

```diff
@@ -356,9 +391,17 @@ export class AssetsContractController extends BaseController<
     selectedAddress: string,
     tokensToDetect: string[],
   ) {
+    // Only fetch balance on networks that support token detection
+    if (!isTokenDetectionEnabledForNetwork(this.config.chainId)) {
+      return {};
+    }
+    const contractAddress =
+      SINGLE_CALL_BALANCES_ADDRESS_BY_CHAINID[
+        this.config.chainId as SupportedTokenDetectionNetworks
```

Comment:
If `chainId` can be any string, then I'm not sure this is right, because it means that `contractAddress` could be undefined. Maybe it's worth doing a `this.config.chainId in SINGLE_CALL_BALANCES_ADDRESS_BY_CHAINID` check here?

Reply:
I added a check right before this using `isTokenDetectionEnabledForNetwork` util and returning {}  if util spits out false. Achieves the same logic but I like your proposed approach better. I've updated this to check `this.config.chainId` in `SINGLE_CALL_BALANCES_ADDRESS_BY_CHAINID` before we grab the `contractAddress` so `contractAddress` should never be undefined. I'm not sure if we can remove the typecasting though since `chainId` is currently type string and `keys` in `SINGLE_CALL_BALANCES_ADDRESS_BY_CHAINID` are of type `SupportedTokenDetectionNetworks`. Any suggestions around this or should we just leave the typecast in this instance?

Reply:
I think in that case I would probably simplify the type on `SINGLE_CALL_BALANCES_ADDRESS_BY_CHAINID` to `Record<string, string>`:

``` typescript
export const SINGLE_CALL_BALANCES_ADDRESS_BY_CHAINID: Record<string, string> = {
  [SupportedTokenDetectionNetworks.mainnet]:
    '0xb1f8e55c7f64d203c1400b9d8555d050f94adf39',
  [SupportedTokenDetectionNetworks.bsc]:
    '0x2352c63A83f9Fd126af8676146721Fa00924d7e4',
  [SupportedTokenDetectionNetworks.polygon]:
    '0x2352c63A83f9Fd126af8676146721Fa00924d7e4',
  [SupportedTokenDetectionNetworks.avax]:
    '0xD023D153a0DFa485130ECFdE2FAA7e612EF94818',
};
```